src/structures/Message.js

```diff
@@ -572,7 +572,7 @@ class Message extends Base {
   get pinnable() {
     return (
       this.type === 'DEFAULT' &&
-      (!this.guild || this.channel.permissionsFor(this.client.user).has(Permissions.FLAGS.MANAGE_MESSAGES, false))
+      (!this.guild || this.channel?.permissionsFor(this.client.user)?.has(Permissions.FLAGS.MANAGE_MESSAGES, false))
     );
   }
```

Comment:
Change the return type to `?boolean` or cast the return to boolean

Reply:
This still returns a boolean

Reply:
I never said it didn't, it also returns `undefined` right now though.

Reply:
No it doesn’t because `boolean && undefined` returns false